kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -221,7 +223,9 @@ private TransactionCommitResult doCommit(
     }
     setTxnOpt.ifPresent(setTxn -> metadataActions.add(createTxnSingleAction(setTxn.toRow())));
 
-    try (CloseableIterator<Row> stageDataIter = dataActions.iterator()) {
+    try (CloseableIterator<Row> stageDataIter =
+        new ValidateDomainMetadataIterator(protocol, dataActions.iterator(), FULL_SCHEMA, false)) {
```

Comment:
Actually I just removed this `boolean allowDuplicate` param, since we should never allow duplicate domainMetadata actions for the same domain. I can’t think of any cases where we’d need this.